src/python/pants/option/config.py

```diff
@@ -406,21 +344,21 @@ def get_value(self, section: str, option: str) -> str | None:
             return remove_val
         return stringify(option_value)
 
-    def options(self, section: str) -> List[str]:
-        section_values = self._find_section_values(section)
+    def options(self, section: str) -> list[str]:
+        section_values = self.values.get(section)
         if section_values is None:
             raise configparser.NoSectionError(section)
-        result = [
+        result = OrderedSet(
```

Comment:
Why is it necessary for this to be a set and it wasn't before?

Reply:
My thinking was make the `default_option not in result` operation faster. But this was probably a premature micro-optimization, and type conversions have a cost too

Reply:
Unless we run `default_option not in result` some non-constant amount of time, then it's O(n), and so is the type conversion. Plus the memory/gc cost. Not that those are significant either. More, let's not attempt to guess at what is faster in cases where it's not obvious.